src/backend/commands/createas.c

```diff
@@ -609,28 +419,239 @@ ExecCreateTableAs(CreateTableAsStmt *stmt, const char *queryString,
 
 		if (into->ivm)
 		{
-
 			Oid matviewOid = address.objectId;
 			Relation matviewRel = table_open(matviewOid, NoLock);
 			Relids	relids = NULL;
 
-			copied_query = copyObject(query);
-			AcquireRewriteLocks(copied_query, true, false);
+			/*
+			 * Mark relisivm field, if it's a matview and into->ivm is true.
+			 */
+			SetMatViewIVMState(matviewRel, true);
 
-			CreateIvmTriggersOnBaseTables(copied_query, (Node *)copied_query->jointree, matviewOid, &relids);
+			if (!into->skipData)
+			{
+				copied_query = copyObject(query);
+				AcquireRewriteLocks(copied_query, true, false);
```

Comment:
This copyObject and AcquireRewriteLocks may be unnecessary as commented as following:
```
 341         /*
 342          * Parse analysis was done already, but we still have to run the rule
 343          * rewriter.  We do not do AcquireRewriteLocks: we assume the query
 344          * either came straight from the parser, or suitable locks were
 345          * acquired by plancache.c.
```
although this is done in the code before this fix, I think this was a mistake.

Reply:
Ok, I will fix it.